tests/infrastructure/vm_console_proxy/utils.py

```diff
@@ -66,3 +70,18 @@ def get_vm_console_proxy_resource(resource_kind, namespace=None):
             name=f"{TOKEN_API_VERSION}.{TOKEN_ENDPOINT}" if resource_kind == APIService else VM_CONSOLE_PROXY
         )
     return vm_console_proxy_resource_object
+
+
+def assert_resource_existence_and_availability(resource: Any, timeout: int) -> None:
+    if resource.kind in {"APIService", "Deployment"}:
+        LOGGER.info(f"Availability/condition check for {resource.kind}.")
+        resource.wait_for_condition(
+            condition=Resource.Condition.AVAILABLE,
+            status=Resource.Condition.Status.TRUE,
+            timeout=timeout,
+        )
+        return
+
+    assert resource.exists, (
```

Comment:
Should this be moved above? If you are not sure if the resource actually exists, then using wait_for_condition() does not seem right.

Reply:
resource.wait_for_condition will check if resource exists and then wait for condition 
if i move assert up , it will e checked for all resources so i will explictly check exists when resource kind != {"APIService", "Deployment"}